app/master/job_config.py

```diff
@@ -18,3 +35,119 @@ def __init__(self, name, setup_build, teardown_build, command, atomizer, max_exe
         self.atomizer = atomizer
         self.max_executors = max_executors
         self.max_executors_per_slave = max_executors_per_slave
+
+    @classmethod
+    def construct_from_dict(cls, name, config_dict):
```

Comment:
Has this discussion with Lu offline. Short answer: I could have gone either way, no big deal either way. I do have a habit of not putting real work in the constructor, just from my experience with non-python languages due to testability concerns.